PhysicsTools/NanoAOD/plugins/NPUTablesProducer.cc

```diff
@@ -0,0 +1,73 @@
+#include "FWCore/Framework/interface/global/EDProducer.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+#include "DataFormats/NanoAOD/interface/FlatTable.h"
+#include "SimDataFormats/GeneratorProducts/interface/LHEEventProduct.h"
+#include "SimDataFormats/PileupSummaryInfo/interface/PileupSummaryInfo.h"
+
+#include <vector>
+#include <iostream>
+
+
+class NPUTablesProducer : public edm::global::EDProducer<> {
```

Comment:
Since you add this, can you remove the instantiation of 
SimplePileupFlatTableProducer from https://github.com/cms-sw/cmssw/blob/master/PhysicsTools/NanoAOD/plugins/SimpleFlatTableProducer.cc#L236 which will no longer be needed?